File: procon7/java/q2/Main.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class Main {
	static Map<Character, Integer> map = new HashMap<>();

	static {
		map.put('I', 1);
		map.put('V', 5);
		map.put('X', 10);
		map.put('L', 50);
		map.put('C', 100);
		map.put('D', 500);
		map.put('M', 1000);
	}

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);

		String s = sc.nextLine();
		int prev = 0;
		int ans = 0;
		int sign = 1;
		for (int i = s.length() - 1; i >= 0; --i) {
			char ch = s.charAt(i);
			int num = map.get(ch);
			if (num > prev) {
				sign = 1;
			} else if (num < prev) {
				sign = -1;
			}
			ans += num * sign;
			prev = num;
		}
		System.out.println(ans);
	}
}

File: procon7/c/q2/Main.c
#include <stdio.h>
#include <string.h>

int main(int argc, char** argv) {
	int map[128];

	char s[1024];

	int prev = 0;
	int ans = 0;
	int num;
	int sign = 1;

	map['I'] = 1;
	map['V'] = 5;
	map['X'] = 10;
	map['L'] = 50;
	map['C'] = 100;
	map['D'] = 500;
	map['M'] = 1000;

	fgets(s, sizeof(s), stdin);

	for (int i = strlen(s) - 1; i >= 0; --i) {
		num = map[s[i]];

		if (num > prev) {
			sign = 1;
		} else if (num < prev) {
			sign = -1;
		}
		ans += num * sign;

		prev = num;
	}

	fprintf(stdout, "%d\n", ans);

	return 0;
}
